### PR TITLE
improvement of TankWithObstacle2D in mprans/SpatialTools.py

### PR DESCRIPTION
**Pull request by lmaurel**
@cekees @adimako 
@tridelat and I made an improvement of TankWithObstacle2D in SpatialTools.py in order to use it for the _submerged_breakwater_ case (https://github.com/erdc-cm/air-water-vv/tree/lmaurel/general_update_review/2d/rubbleMoundBreakWater/Submerged_breakwater). Basically it permits to create a segment at the bottom of the obstacle and to fill the obstacle with what we want (for example a porous zone in the submerged breakwater). This won't affect the other cases where we already use TankWithObstacle2D .

### CODE REVIEW

**Comment by cekees** [comment]
+1

**Comment by adimako** [comment]
@lmaurel I understand that you have discussed this with @tridelat, as well so you are good to go

File: proteus/mprans/SpatialTools.py [hunk 1]
```diff
@@ -1206,7 +1206,9 @@ def __init__(self, domain, dim=(0., 0.),
 
         self.corners = {'x-y-': False, 'x+y-': False,
                         'x+y+': False, 'x-y+': False}
-
+                        
+	self.hole = hole
```
Comment (alistairbntl) [inline proteus/mprans/SpatialTools.py]: The indentation looks odd for line 1210.  Do you need an extra space here?

File: proteus/mprans/SpatialTools.py [hunk 2]
```diff
@@ -1186,7 +1186,7 @@ class TankWithObstacles2D(Tank2D):
     def __init__(self, domain, dim=(0., 0.),
                  obstacles = None, special_boundaries = None,
                  full_circle = False,
-                 coords=None, from_0=True):
+                 coords=None, from_0=True, hole=True, obstacle_regions=None):
```
Comment (alistairbntl) [inline proteus/mprans/SpatialTools.py]: Can you update the comments to reflect the new input arguments?

File: proteus/mprans/SpatialTools.py (hunk 1 above)
Comment (lmaurel) [inline proteus/mprans/SpatialTools.py]: You're right, I'll fix this. Thanks

File: proteus/mprans/SpatialTools.py (hunk 2 above)
Comment (lmaurel) [inline proteus/mprans/SpatialTools.py]: Yes I'm going to do it.

**Pull request merged**